cmd/kubeadm/app/cmd/token_test.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"flag"
+	"regexp"
+	"testing"
+
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+	TokenExpectedRegex = "^\\S{6}\\.\\S{16}\n$"
+)
+
+var kubeadmPath string
+
+func init() {
+	flag.StringVar(&kubeadmPath, "kubeadm-path", "", "Location of kubeadm")
```

Comment:
This seems much more like integration tests than unit tests to me.

Do everyone think that we should have integration tests yet?
Just wondering, because they probably should be in another package totally, and share this logic.

Because we don't want to have every `_test.go` file to have this `kubeadm-path` logic in it, and it doesn't really feel ideomatic what we do below:
```
make -C "${KUBE_ROOT}" test \
  WHAT=cmd/kubeadm/app/cmd \
  KUBE_TEST_ARGS="-run Cmd --kubeadm-path '${KUBEADM_PATH}'"
```

It would be much better if you could split this logic into a separate pkg; initially maybe `cmd/kubeadm/integration` or something.

And I expect unit tests (i.e. `token_test.go`, `init_test.go`, etc.) to call the functions of the main file directly (i.e. `token.go` and `init.go`) like this:
```go
func TestRunGenerateToken (t *testing.T) {
   // init mywriter
   err := RunGenerateToken(mywriter)
   // parse output written to mywriter
   // etc...
}
```